File: CousinsinBinaryTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isCousins(TreeNode* root, int x, int y) {
        if(!root) return 0;
        queue<TreeNode*>q;
        q.push(root);
        while(!q.empty())
        {
            int size=q.size();
            bool A_exist=false;
            bool B_exist=false;
            for(int i=0;i<size;i++)
            {
                TreeNode *temp=q.front();
                q.pop();
                if(temp->val==x) A_exist=true;
                if(temp->val==y) B_exist=true;
                if(temp->left!=NULL and temp->right!=NULL)
                {
                    if(temp->left->val==x and temp->right->val==y)
                    {
                        return false;
                    }
                    if(temp->left->val==y and temp->right->val==x) return false;
                }
                if(temp->left!=NULL) q.push(temp->left);
                if(temp->right!=NULL) q.push(temp->right);
            }
            if(A_exist and B_exist) return true;
        }
        return false;
    }
};

File: cousins-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    x_depth,y_depth=0,0
    a,b=None,None
    def isCousins(self, root: TreeNode, x: int, y: int) -> bool:
        depth=0
        self.dfs(root,x,y,depth,root)
        #Solutin.x_depth,Solutioy_depth
        if(Solution.x_depth==Solution.y_depth and Solution.a!=Solution.b):
            return 1
        else:
            return 0
    def dfs(self,root,x,y,depth,parent):
        if not root:
            return
        else:
            depth+=1
            if(root.val==x):
                Solution.x_depth=depth
                Solution.a=parent
            if(root.val==y):
                Solution.y_depth=depth
                Solution.b=parent
            self.dfs(root.left,x,y,depth,root)
            self.dfs(root.right,x,y,depth,root)
